### ISSUE TITLE
Installation error: identifier "src_zeropad" is undefined

### ISSUE THREAD

**Issue by wangjia-ai** [ECP-WarpX/WarpX#2347]
Dear developers:
        Recently, another installation error happens as seen below
        
![error2](https://user-images.githubusercontent.com/73467797/134878766-fd2a8f6d-16a9-457b-b24c-a922f8a817c0.png)
       Thanks

**Comment by ax3l** [comment]
@wangjia-ai thank you for the report!
Can you attach the full build output when you start this from a fresh, empty `build` directory? I wonder if you GCC is too old or not matching the nvcc version or if it's another problem.

**Comment by ax3l** [comment]
Also seen by @PhilMiller

**Comment by PhilMiller** [comment]
`cmake ~/repos/WarpX/  -DWarpX_DIMS=2 -DWarpX_EB=ON -DWarpX_LIB=ON -DWarpX_COMPUTE=OMP -DWarpX_OPENPMD=ON -G Ninja`

```
-- Found CCache: /usr/bin/ccache
-- Downloading AMReX ...
-- AMReX repository: https://github.com/AMReX-Codes/amrex.git (350a3efc2c6408efcadb6aaf0a334944af51098b)
-- Enabled CUDA options:
--    AMReX_CUDA_FASTMATH
--    AMReX_CUDA_MAX_THREADS = 256
--    AMReX_CUDA_MAXREGCOUNT = 255
--    AMReX_CUDA_WARN_CAPTURE_THIS
--    AMReX_CUDA_SHOW_LINENUMBERS
--    AMReX_CUDA_SHOW_CODELINES
-- Autodetected CUDA architecture(s):  7.0
-- CMake version: 3.16.3
-- AMReX installation directory: /usr/local
-- Build type set by user to 'Release'.
-- Building AMReX with AMReX_SPACEDIM = 2
-- Configuring AMReX with the following options enabled: 
--    AMReX_PRECISION = DOUBLE
--    AMReX_GPU_BACKEND = CUDA
--    AMReX_MPI
--    AMReX_MPI_THREAD_MULTIPLE
--    AMReX_EB
--    AMReX_LINEAR_SOLVERS
--    AMReX_PARTICLES
--    AMReX_PARTICLES_PRECISION = DOUBLE
--    AMReX_PIC
--    AMReX_TINY_PROFILE
-- Found MPI: TRUE (found version "3.1") found components: C CXX 
-- AMReX configuration summary: 
--    Build type               = Release
--    Install directory        = /usr/local
--    C++ compiler             = /usr/bin/c++
--    CUDA compiler            = /usr/local/cuda-11.3/bin/nvcc
--    C++ defines              = 
--    C++ flags                = -O3 -DNDEBUG  -pthread -Werror=return-type
--    CUDA flags               = -O3 -DNDEBUG  -m64 --expt-relaxed-constexpr --expt-extended-lambda -Wno-deprecated-gpu-targets -gencode=arch=compute_70,code=sm_70 -maxrregcount=255 -Xcudafe --diag_suppress=esa_on_defaulted_function_ignored --use_fast_math -Xcudafe --display_error_number --Wext-lambda-captures-this --forward-unknown-to-host-compiler --generate-line-info --source-in-ptx --display-error-number --diag-error 20092 -Xcompiler -pthread
--    C++ include paths        = -I/home/ubuntu/build/2021-09-02-cuda/_deps/fetchedamrex-src/Src/Base -I/home/ubuntu/build/2021-09-02-cuda/_deps/fetchedamrex-src/Src/Base/Parser -I/home/ubuntu/build/2021-09-02-cuda/_deps/fetchedamrex-src/Src/Boundary -I/home/ubuntu/build/2021-09-02-cuda/_deps/fetchedamrex-src/Src/AmrCore -I/home/ubuntu/build/2021-09-02-cuda/_deps/fetchedamrex-src/Src/EB -I/home/ubuntu/build/2021-09-02-cuda/_deps/fetchedamrex-src/Src/LinearSolvers/MLMG -I/home/ubuntu/build/2021-09-02-cuda/_deps/fetchedamrex-src/Src/LinearSolvers/Projections -I/home/ubuntu/build/2021-09-02-cuda/_deps/fetchedamrex-src/Src/Particle -I/usr/lib/x86_64-linux-gnu/openmpi/include/openmpi -I/usr/lib/x86_64-linux-gnu/openmpi/include
--    Link line                = -pthread /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi_cxx.so /usr/local/cuda-11.3/targets/x86_64-linux/lib/libnvToolsExt.so
-- AMReX: Using version '21.09' (21.09-32-g350a3efc2c64)
-- Downloading PICSAR ...
-- PICSAR repository: https://github.com/ECP-WarpX/picsar.git (a78be127f66adc1558f527edc8964e37e3a055ff)
-- Downloading openPMD-api ...
-- openPMD-api repository: https://github.com/openPMD/openPMD-api.git (0.14.2)
-- Found MPI: TRUE (found version "3.1") found components: CXX 
-- Using the single-header code from /home/ubuntu/build/2021-09-02-cuda/_deps/fetchedopenpmd-src/share/openPMD/thirdParty/json/single_include/
-- nlohmann-json: Using INTERNAL version '3.9.1'
-- HDF5: Using hdf5 compiler wrapper to determine C configuration
-- Can NOT find 'adios_config' - set ADIOS_ROOT, ADIOS_DIR or INSTALL_PREFIX, or check your PATH
-- Could NOT find ADIOS (missing: ADIOS_LIBRARIES ADIOS_INCLUDE_DIRS) (Required is at least version "1.13.1")
-- Could NOT find ADIOS2 (missing: ADIOS2_DIR)
-- MPark.Variant: Using INTERNAL version '1.4.0'

openPMD build configuration:
  library Version: 0.14.2
  openPMD Standard: 1.1.0
  C++ Compiler: GNU 9.3.0 
    /usr/bin/c++

  Installation: OFF

  Build Type: Release
  Library: static
  CLI Tools: OFF
  Examples: OFF
  Testing: OFF
  Invasive Tests: OFF
  Internal VERIFY: ON
  Build Options:
    MPI: ON
    HDF5: ON
    ADIOS1: OFF
    ADIOS2: OFF
    PYTHON: OFF


WarpX build configuration:
  Version: 21.9 (21.09-174-g33023a0cf237-dirty)
  C++ Compiler: GNU 9.3.0 
    /usr/bin/c++

  Installation prefix: /usr/local
        bin: bin
        lib: lib
    include: include
      cmake: lib/cmake/WarpX

  Build type: Release
  Build options:
    APP: ON
    ASCENT: OFF
    COMPUTE: CUDA
    DIMS: 2
    Embedded Boundary: ON
    GPU clock timers: ON
    IPO/LTO: OFF
    LIB: ON (static)
    MPI: ON
    PSATD: OFF
    PRECISION: DOUBLE
    OPENPMD: ON
    QED: ON
    QED table generation: OFF
    SENSEI: OFF

-- Configuring done
-- Generating done
-- Build files have been written to: /home/ubuntu/build/2021-09-02-cuda
```

**Comment by ax3l** [comment]
Uff, yes should not compile with GPU for all but 3D cases.

Regression to #2314 since we only compile in 3D with CUDA/SYCL/HIP atm (due to CI time).
Will add a 2D compilation to HIP.

**Comment by ax3l** [comment]
Thank you both for the report!
Fix in #2353 :)
